sdk/keyvault/azure-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretAsyncClient.java

```diff
@@ -466,17 +607,36 @@
      * restored secret details when a response is received.</p>
      * <pre>
      * //Pass the Secret Backup Byte array to the restore operation.
-     * secretAsyncClient.restoreSecret(secretBackupByteArray).subscribe(secretResponse -&gt;
-     *   System.out.printf("Restored Secret with name %s and value %s \n", secretResponse.value().name(), secretResponse.value().value()));
-     * </pre>
+     * {@codesnippet com.azure.keyvault.secrets.secretclient.restoreSecret#byte}
+     *
+     * @param backup The backup blob associated with the secret.
+     * @return A {@link Mono} containing the {@link Secret restored secret}.
+     * @throws ResourceModifiedException when {@code backup} blob is malformed.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Secret> restoreSecret(byte[] backup) {
+        return this.restoreSecretWithResponse(backup).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Restores a backed up secret, and all its versions, to a vault. This operation requires the
+     * {@code secrets/restore} permission.
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Restores the secret in the key vault from its backup. Subscribes to the call asynchronously
+     * and prints out the
+     * restored secret details when a response is received.</p>
+     * <pre>
+     * //Pass the Secret Backup Byte array to the restore operation.
+     * {@codesnippet com.azure.keyvault.secrets.secretclient.restoreSecretWithResponse#byte}
      *
      * @param backup The backup blob associated with the secret.
      * @return A {@link Mono} containing a {@link Response} whose {@link Response#value() value}
      * contains the {@link Secret restored secret}.
      * @throws ResourceModifiedException when {@code backup} blob is malformed.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<Secret>> restoreSecret(byte[] backup) {
+    public Mono<Response<Secret>> restoreSecretWithResponse(byte[] backup) {
         return withContext(context -> restoreSecret(backup, context));
```

Comment:
The private methods should be named with the `withResponse` suffix, if that is what they return.